src/Repository/EmailSubscriptionHistoryRepository.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Reporting\EmailSubscriptionHistory;
+use AppBundle\Entity\Reporting\EmailSubscriptionHistoryAction;
+use Carbon\Carbon;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Common\Persistence\ManagerRegistry;
+
+class EmailSubscriptionHistoryRepository extends ServiceEntityRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, EmailSubscriptionHistory::class);
+    }
+
+    /**
+     * Count active adherents (not users) history for each specified subscription types in the referent managed area before the specified date.
+     */
+    public function countAllByTypeForReferentManagedArea(Adherent $referent, array $subscriptionsTypes, \DateTime $until): array
```

Comment:
what about rename this method on `...ForReferentTags(array $referentTags...` and remove the `!$referent->isReferent()` check?
because it's what you do, filter by tag (it can be the tags of one referent or of one simple adherent)?

Reply:
I do agree :+1: